distribution/lib/Standard/Test/0.0.0-dev/src/Frame_Hider.enso

```diff
@@ -0,0 +1,118 @@
+from Standard.Base import all
+import Standard.Base.Runtime.Ref.Ref
+
+## PRIVATE
+
+   Frame_Hider is used in test utilities to hide uninteresting stack frames from
+   users when reporting the location of test failures.
+
+   In a test utility such as `Any.should_equal`, the failure of a test assertion
+   is accompanied by the location of the failure. The location should not be a
+   stack frame inside `Any.should_equal`, but rather should be at the location
+   that called `Any.should_equal`. This means identifying stack frames we don't
+   want to bother the user with, and omitting them when we're looking for the
+   correct stack frame to use.
+
+   The problem is more complicated when test utilities call other utilities, or
+   when test utilities invoke callbacks provided by user test code.
+
+   The `hide` and `unhide*` methods are simple wrappers. `get_top_stack_frame`
+   identifies these on the stack, and uses them to removes the uninteresting
+   frames from consderation.
+
+   A `hide` call should be wrapped around the body of every test utility.
+
+   An `unhide` call should be wrapped around any call from a test utility to a
+   user-supplied callback, depending on the number of arguments. If one test
+   utility forwards a callback to another test utility, the callback should be
+   wrapped at the forward site as well.
+type Frame_Hider
+    ## PRIVATE
+
+       Hides the stack frames of the contained code from `get_top_stack_frame`.
+
+       > Example
+         Wrap the body of a test utility to hide its stack frames.
+
+         Any.should_equal self that = Frame_Hider.hide <|
+             ...
+             Error.throw ("Not equal at " + Frame_Hider.get_top_stack_frame)
+    hide ~action =
+        action
+
+    ## PRIVATE
+
+       Unhides the stack frames of the call to the user-supplied callback
+       application.
+
+       > Example
+         Wrap a callback invocation to unhide its stack frames.
+
+         my_test_utility test_predicate x =
+             ...
+             Frame_Hider.unhide (test_predicate x)
+    unhide ~action =
+        action
+
+    ## PRIVATE
+
+       Unhides the stack frames of the call to the user-supplied callback
+       application, uncurried for one argument.
+
+       `unhide_1 f x` is Equivalent to `unhide (f x)`.
+
+       > Example
+         Wrap a callback invocation to unhide its stack frames.
+
+         my_test_utility test_predicate x =
+             ...
+             Frame_Hider.unhide_1 test_predicate x
+    unhide_1 f x =
+        f x
```

Comment:
When should I prefer to call `unhide_1 f x` versus `unhide (f x)`. Is there any difference? What is it?